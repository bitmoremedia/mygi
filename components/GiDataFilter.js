import React from 'react';

import PageLink from './common/PageLink';
import { categoryPathMap, pathCategoryMap } from '../utils';

const GiDataFilter = ({categoryFilter}) => {

  const FilterLink = ({categoryPath}) => {

    let isActive = false;
    const category = (categoryPath) ? pathCategoryMap()[categoryPath] : 'All';

    if (!categoryFilter && !categoryPath) {
      isActive = true;
    } else {
      if ( category === categoryFilter ) {
        isActive = true;
      }
    }

    const pathTo = (categoryPath) ? '/glycemic-index/' + categoryPath : '/glycemic-index/';

    return (
      <div>
        <PageLink to={pathTo}>
          {category} {isActive && '*'}
        </PageLink>
      </div>
    );
  };

  return (
    <div className="gi-data-filter">
      <FilterLink/>
      <FilterLink categoryPath="beans"/>
      <FilterLink categoryPath="breads"/>
      <FilterLink categoryPath="breakfast-cereals"/>
      <FilterLink categoryPath="dairy"/>
      <FilterLink categoryPath="fruits"/>
      <FilterLink categoryPath="snacks-and-sweet-foods"/>
      <FilterLink categoryPath="staples"/>
      <FilterLink categoryPath="vegetables"/>
    </div>
  );
};

export default GiDataFilter;
