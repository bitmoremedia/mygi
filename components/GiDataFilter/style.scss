@import '../../css/variables';

.gi-data-filter {
  padding-bottom: 10px;
  text-align: center;
  @include respond-to(tablet) {
    padding-top: 10px;
  }
}
.gi-data-filter h2 {
  font-size: 14px;
  letter-spacing: inherit;
  text-transform: uppercase;
  margin: 5px 0px 15px 0px;
}
.gi-data-filter__filter-item {
  height: 2.8rem;
  line-height: 2.8rem;
  padding: 0 1.5rem;
  margin-right: 5px;
}
.gi-data-filter__filter-item--active,
.gi-data-filter__filter-item--active:hover {
  border-color: $main;
  background-color: $main;
}
