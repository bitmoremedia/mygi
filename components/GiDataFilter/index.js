import React from 'react';
import classNames from 'classnames';

import './style';
import PageLink from '../common/PageLink';
import { categoryPathMap, pathCategoryMap } from '../../utils';

const GiDataFilter = ({categoryFilter, giTypeFilter}) => {

  const currentCategory = categoryFilter;
  const currentCategoryPath = categoryPathMap()[categoryFilter];
  const currentGiTypeFilter = giTypeFilter;
  const currentGiTypePath = (currentGiTypeFilter) ? `${giTypeFilter}/` : ``;

  const FilterLink = ({categoryPath, giType}) => {

    let pathTo, linkTitle, isActive = false;

    // category filter links
    if (categoryPath) {
      pathTo = (categoryPath === 'all') ? `/glycemic-index/${currentGiTypePath}` : `/glycemic-index/${currentGiTypePath}${categoryPath}`;
      linkTitle = (categoryPath === 'all') ? 'All' : pathCategoryMap()[categoryPath];
      if ( currentCategoryPath === categoryPath || ( !currentCategoryPath && categoryPath === 'all' ) ){
        isActive = true;
      }
    }

    // gi type filter links
    if (giType) {
      isActive = (currentGiTypeFilter === giType);
      //debugger;
      pathTo = (currentCategoryPath) ? `/glycemic-index/${giType}/${currentCategoryPath}` : `/glycemic-index/${giType}`;
      switch (giType) {
        case 'all':
          linkTitle = 'All';
          pathTo = (currentCategoryPath) ? `/glycemic-index/${currentCategoryPath}` : `/glycemic-index`;
          isActive = ( isActive || ! currentGiTypeFilter);
          break;
        case 'low-gi':
          linkTitle = 'Low'
          break;
        case 'medium-gi':
          linkTitle = 'Medium'
          break;
        case 'high-gi':
          linkTitle = 'High'
          break;
        default:
          linkTitle = 'All'
      }
    }

    const linkClass = classNames('gi-data-filter__filter-item button', { 'button-outline' : !isActive, 'gi-data-filter__filter-item--active' : isActive });

    return (
      <PageLink className={linkClass} to={pathTo}>
        {linkTitle}
      </PageLink>
    );
  };

//       <div className="gi-data-filter__divider"></div>

  return (
    <div className="gi-data-filter">
      <h2>Category</h2>
      <FilterLink categoryPath="all"/>
      <FilterLink categoryPath="beans"/>
      <FilterLink categoryPath="breads"/>
      <FilterLink categoryPath="breakfast-cereals"/>
      <FilterLink categoryPath="dairy"/>
      <FilterLink categoryPath="fruits"/>
      <FilterLink categoryPath="snacks-and-sweet-foods"/>
      <FilterLink categoryPath="staples"/>
      <FilterLink categoryPath="vegetables"/>
      <h2>GI Type</h2>
      <FilterLink giType="all"/>
      <FilterLink giType="low-gi"/>
      <FilterLink giType="medium-gi"/>
      <FilterLink giType="high-gi"/>
    </div>
  );
};

export default GiDataFilter;
