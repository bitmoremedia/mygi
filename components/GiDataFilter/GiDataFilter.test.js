import React from 'react'
import renderer from 'react-test-renderer'

import GiDataFilter from './index'

test('default filters display with the two "All" buttons as active', () => {
  const component = renderer.create(
    <GiDataFilter />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('category "Beans" filter shows relevant button as active', () => {
  const component = renderer.create(
    <GiDataFilter categoryFilter="Beans" />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('gi type "Low GI" filter shows relevant button as active', () => {
  const component = renderer.create(
    <GiDataFilter giTypeFilter="low-gi" />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('combined filters "Beans" and "Low GI" show the relevant buttons as active', () => {
  const component = renderer.create(
    <GiDataFilter categoryFilter="Beans" giTypeFilter="low-gi" />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
