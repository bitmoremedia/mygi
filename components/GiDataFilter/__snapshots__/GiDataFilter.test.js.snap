// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category "Beans" filter shows relevant button as active 1`] = `
<div
  className="gi-data-filter"
>
  <h2>
    Category
  </h2>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    All
  </a>
  <a
    className="gi-data-filter__filter-item button gi-data-filter__filter-item--active"
    onClick={[Function]}
    style={Object {}}
  >
    Beans
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Breads
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Cereals
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Dairy
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Fruits
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Snacks & Sweets
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Staples
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Vegetables
  </a>
  <h2>
    GI Type
  </h2>
  <a
    className="gi-data-filter__filter-item button gi-data-filter__filter-item--active"
    onClick={[Function]}
    style={Object {}}
  >
    All
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Low
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Medium
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    High
  </a>
</div>
`;

exports[`combined filters "Beans" and "Low GI" show the relevant buttons as active 1`] = `
<div
  className="gi-data-filter"
>
  <h2>
    Category
  </h2>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    All
  </a>
  <a
    className="gi-data-filter__filter-item button gi-data-filter__filter-item--active"
    onClick={[Function]}
    style={Object {}}
  >
    Beans
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Breads
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Cereals
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Dairy
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Fruits
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Snacks & Sweets
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Staples
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Vegetables
  </a>
  <h2>
    GI Type
  </h2>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    All
  </a>
  <a
    className="gi-data-filter__filter-item button gi-data-filter__filter-item--active"
    onClick={[Function]}
    style={Object {}}
  >
    Low
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Medium
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    High
  </a>
</div>
`;

exports[`default filters display with the two "All" buttons as active 1`] = `
<div
  className="gi-data-filter"
>
  <h2>
    Category
  </h2>
  <a
    className="gi-data-filter__filter-item button gi-data-filter__filter-item--active"
    onClick={[Function]}
    style={Object {}}
  >
    All
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Beans
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Breads
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Cereals
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Dairy
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Fruits
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Snacks & Sweets
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Staples
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Vegetables
  </a>
  <h2>
    GI Type
  </h2>
  <a
    className="gi-data-filter__filter-item button gi-data-filter__filter-item--active"
    onClick={[Function]}
    style={Object {}}
  >
    All
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Low
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Medium
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    High
  </a>
</div>
`;

exports[`gi type "Low GI" filter shows relevant button as active 1`] = `
<div
  className="gi-data-filter"
>
  <h2>
    Category
  </h2>
  <a
    className="gi-data-filter__filter-item button gi-data-filter__filter-item--active"
    onClick={[Function]}
    style={Object {}}
  >
    All
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Beans
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Breads
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Cereals
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Dairy
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Fruits
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Snacks & Sweets
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Staples
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Vegetables
  </a>
  <h2>
    GI Type
  </h2>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    All
  </a>
  <a
    className="gi-data-filter__filter-item button gi-data-filter__filter-item--active"
    onClick={[Function]}
    style={Object {}}
  >
    Low
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    Medium
  </a>
  <a
    className="gi-data-filter__filter-item button button-outline"
    onClick={[Function]}
    style={Object {}}
  >
    High
  </a>
</div>
`;
